warp/verifier_backend.go

```diff
@@ -62,3 +66,70 @@ func (b *backend) verifyBlockMessage(ctx context.Context, blockHashPayload *payl
 
 	return nil
 }
+
+// verifyAddressedCall verifies the addressed call message
+func (b *backend) verifyAddressedCall(addressedCall *payload.AddressedCall) *common.AppError {
```

Comment:
Should we rename this `verifyOffChainAddressedCall`? Since all addressed calls emitted on-chain will be verified already above by calling `GetMessage`.

Also, I think we should add the requirement that `len(addressedCall.SourceAddress) == 0` here, so that off-chain messages can always be differentiated from messages emitted on-chain. Otherwise, if a contract wanted to emit a message with a format matching one defined here, a message from that contract could be "forged".

Reply:
I see now that `verifyUptimeMessage` has the check on `len(sourceAddress)` so the code is correct, though still think it likely would be a bit less error prone to include that check here instead so it applies in all cases for off-chain messages.